core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -2425,6 +2449,381 @@ public static Object structAccess(Object structObject, int index, String fieldNa
+          return PathContext.withStrictException((Exception) input);
+        }
+        ctx = JsonPath.parse(input,
+            Configuration
+                .builder()
+                .jsonProvider(JSON_PATH_JSON_PROVIDER)
+                .mappingProvider(JSON_PATH_MAPPING_PROVIDER)
+                .build()
+        );
+        break;
+      case LAX:
+        if (input instanceof Exception) {
+          return PathContext.withReturned(PathMode.LAX, null);
+        }
+        ctx = JsonPath.parse(input,
+            Configuration
+                .builder()
+                .options(Option.SUPPRESS_EXCEPTIONS)
+                .jsonProvider(JSON_PATH_JSON_PROVIDER)
+                .mappingProvider(JSON_PATH_MAPPING_PROVIDER)
+                .build()
+        );
+        break;
+      default:
+        throw new RuntimeException("illegal mode: " + mode);
+      }
+      try {
+        return PathContext.withReturned(mode, ctx.read(pathWff));
+      } catch (Exception e) {
+        return PathContext.withStrictException(e);
+      }
+    } catch (Exception e) {
+      return PathContext.withUnknownException(e);
+    }
+  }
+
+  public static Boolean jsonExists(Object input) {
+    return jsonExists(input, SqlJsonExistsErrorBehavior.FALSE);
+  }
+
+  public static Boolean jsonExists(Object input, SqlJsonExistsErrorBehavior errorBehavior) {
+    PathContext context = (PathContext) input;
+    if (context.exc != null) {
+      switch (errorBehavior) {
+      case TRUE:
+        return Boolean.TRUE;
+      case FALSE:
+        return Boolean.FALSE;
+      case ERROR:
+        throw new RuntimeException(context.exc);
+      case UNKNOWN:
+        return null;
+      default:
+        throw new RuntimeException("illegal error behavior: " + errorBehavior);
+      }
+    } else {
+      return !Objects.isNull(context.pathReturned);
+    }
+  }
+
+  public static Object jsonValueAny(Object input,
+                                 SqlJsonValueEmptyOrErrorBehavior emptyBehavior,
+                                 Object defaultValueOnEmpty,
+                                 SqlJsonValueEmptyOrErrorBehavior errorBehavior,
+                                 Object defaultValueOnError) {
+    PathContext context = (PathContext) input;
+    Exception exc;
+    if (context.exc != null) {
+      exc = context.exc;
+    } else {
+      Object value = context.pathReturned;
+      if (value == null || context.mode == PathMode.LAX
+          && !isScalarObject(value)) {
+        switch (emptyBehavior) {
+        case ERROR:
+          throw new RuntimeException("empty json value");
+        case NULL:
+          return null;
+        case DEFAULT:
+          return defaultValueOnEmpty;
+        default:
+          throw new RuntimeException("illegal empty behavior: " + emptyBehavior);
+        }
+      } else if (context.mode == PathMode.STRICT && !isScalarObject(value)) {
+        exc = new RuntimeException("not a json value: " + value);
```

Comment:
I wonder if we should use `RuntimeExceptions` for errors like this.
Should `CalciteResource` be used?
Should the message help user more to understand and resolve the failure?

E.g. "strict jsonpath mode requires scalar value, and the actual value is: ..."

Reply:
If we use CalciteResource we should factor the previous usages of RuntimeException in other function code too, to keep consistency. If it is OK I can do the work in this PR.